### PR TITLE
Added a 'peek' tool for guessing best array model

### PR DESCRIPTION
**Pull request by kvg**
Added the 'peek' subcommand to determine the most likely array model to which data from a MAS-ISO-seq experiment conforms.

Also prints a histogram of the counts, which is helpful for debugging and determining if data comes from a multiplexed experiment.

### CODE REVIEW

File: src/longbow/peek/command.py [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+    # Check to see if the output files exist:
+    bam_utils.check_for_preexisting_files(output_model, exist_ok=force)
+
+    threads = mp.cpu_count() if threads <= 0 or threads > mp.cpu_count() else threads
+    logger.info(f"Running with {threads} worker subprocess(es)")
+
+    # Make all prebuilt models
+    models = {}
+    for model_name in LibraryModel.pre_configured_models:
+        if not LibraryModel.pre_configured_models[model_name]['deprecated']:
```
Comment (jonn-smith) [inline src/longbow/peek/command.py:124]: Can you either add an argument to include deprecated models or a warning here for each deprecated model it skips?

File: src/longbow/peek/command.py [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+        results.put(None)
+        output_worker.join()
+
+    logger.info("Histogram of most likely model counts:")
+
+    res = dict(sorted(res.items(), key=lambda item: item[1], reverse=True))
+    plot_model_counts(res)
+    best_model = next(iter(res.keys()))
+
+    logger.info(f"Overall most likely model: {best_model}")
```
Comment (jonn-smith) [inline src/longbow/peek/command.py:232]: Can you add a statistic to go with this log statement?  Seems like you can pull it directly from the histogram.

File: src/longbow/peek/command.py [hunk 3]
```diff
@@ -0,0 +1,360 @@
...
+def plot_model_counts(res, max_width=50.0):
+    block_char = u"\u2588"
+
+    model_tot = np.sum(list(res.values()))
+    max_label_width = np.max(list(map(lambda x: len(x), res.keys())))
+    for model_name in res:
+        pct = 100.0 * res[model_name] / model_tot
+        num_blocks = int(res[model_name] * max_width / model_tot)
+
+        logger.info(f"  {model_name:>{max_label_width}} {block_char*(num_blocks+1)} {res[model_name]} ({pct:.1f}%)")
```
Comment (jonn-smith) [inline src/longbow/peek/command.py:251]: Nice use of unicode!

File: src/longbow/peek/command.py [hunk 4]
```diff
@@ -0,0 +1,360 @@
...
+
+        model_scores[f'{model_name}:fw'] = fw_logp
+        model_scores[f'{model_name}:rc'] = rc_logp
+
+    model_scores = dict(sorted(model_scores.items(), key=lambda item: item[1], reverse=True))
+
+    best_model_full = next(iter(model_scores))
+    best_model = re.sub(':.*$', '', best_model_full)
+
+    return best_model
```
Comment (jonn-smith) [inline src/longbow/peek/command.py:360]:
```suggestion
    return best_model
    
```

**Review by jonn-smith — CHANGES_REQUESTED** [review]
Looks great!  A couple minor requests

File: src/longbow/peek/command.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (kvg) [reply src/longbow/peek/command.py]: Done!

File: src/longbow/utils/model.py [hunk 5]
```diff
@@ -237,14 +237,15 @@ def build(self):
         self._initialize_hmm_transitions_new()
 
         # DEBUGGING:
-        self.dump_as_dotfile(do_subgraphs=False)
-        self.dump_as_dotfile_simple()
-        self.to_json(f"longbow_model_{self.name}.v{self.version}.json")
+        if logger.isEnabledFor(logging.DEBUG):
```
[the remark this answers is not in the document]
Comment (kvg) [reply src/longbow/utils/model.py:244]: Thanks!

File: src/longbow/utils/model.py [hunk 6]
```diff
@@ -1333,6 +1334,7 @@ def build_pre_configured_model(model_name):
             "named_random_segments": None,
             "coding_region": None,
             "annotation_segments": None,
+            "deprecated": True,
```
[the remark this answers is not in the document]
Comment (kvg) [reply src/longbow/utils/model.py:1341]: Added.

File: src/longbow/peek/command.py (hunk 2 above)
Comment (kvg) [reply src/longbow/peek/command.py]: Added.

File: src/longbow/peek/command.py (hunk 3 above)
Comment (kvg) [reply src/longbow/peek/command.py]: Thanks!

**Review by jonn-smith — APPROVED** [review]

**Pull request merged**